microprofile/WFLY-17137-microprofile-jakarta-ee-10.adoc

```diff
@@ -0,0 +1,185 @@
+
+The below table shows the versions of each specification in MicroProfile Platform 5, the current versions included in WildFLy, and the planned version for MicroProfile Platform 6. Only the minor versions are included, we will use the latest micro versions of each at the time of release. It also contains a column indicating whether something changes the functionality provided by a specification.
+
+[cols="1,1,1,1,1"]
+|===
+|Spec | MP 5 Version | Current WildFly Version | MP 6 Version | Updated Functionality (y/n)
+
+|Config
+|3.0
+|3.0
+|3.0
+|n
+
+|Fault Tolerance
+|4.0
+|4.0
+|4.0
+|n
+
+|Health
+|4.0
+|4.0
+|4.0
+|n
+
+|JWT
+|2.0
+|2.0
+|TODO: 2.0 (might become 2.1)
+|TODO: ??? (depends on 2.1)
+
+|+++<s>Metrics</s>+++ +
+Will not include
+|+++<s>4.0</s>+++
+|+++<s>4.0</s>+++
+|+++<s>5.0</s>+++
+|+++<s>y</s>+++
+
+|OpenAPI
+|3.0
+|3.0
+|3.1
+|TODO: ??
+
+|OpenTracing +
+Dropped in MP 6
+|3.0
+|3.0
+|-
+|-
+
+|Rest Client
+|3.0
+|3.0
+|3.0
+|n
+
+|Telemetry +
+New in MP 6
+|-
+|-
+|1.0
+|y -
+
+|===
+
+TODO: More about MP Telemetry - will we support it? If we are not that needs to be pointed out. If we are we need links to its RFE.
+
+
+== Issue Metadata
+
+* https://issues.redhat.com/browse/WFLY-17137[WFLY-17137] - Provide Support for MicroProfile.Next version that Aligns with Jakarta EE 10
+* https://issues.redhat.com/browse/EAP7-1861[EAP7-1861] - Provide Support for MicroProfile.Next version that Aligns with Jakarta EE 10
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE[WFCORE-XXXX]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-17138[WFLY-17138] - Convert MP Metrics subsystem to Legacy Mode
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+WildFly will be updated to pull in the corresponding versions of the Eclipse Microprofile specification jars, as well as the SmallRye jars and dependencies needed to implement each version.
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* All components will be upgraded to the latest micro versions of versions of the Eclipse Microprofile specification jars, as well as the SmallRye jars and dependencies needed to implement each version, as indicated in the table in the link:#overview[Overview].
+** Upgrading the components will provide the latest TCKs
+* The MicroProfile Metrics subsystem will be converted to 'legacy mode', containing a `migrate` operation to remove it.
```

Comment:
Remove it or migrate it to Micrometer or MP Telemetry?

Also, I've forgotten if we have any other cases of a 'migrate' op that is nothing more than a self-remove. If not, let's not introduce that concept just for this, at least not without a broader discussion (i.e. not just people interested in this work.)

Reply:
I'll have to research that operation before I can make an intelligent comment here. :)